staging/src/k8s.io/apiserver/pkg/server/routes/openapi.go

```diff
@@ -51,3 +55,36 @@ func (oa OpenAPI) Install(c *restful.Container, mux *mux.PathRecorderMux) (*hand
 
 	return openAPIVersionedService, spec
 }
+
+func (oa OpenAPI) InstallV3(c *restful.Container, mux *mux.PathRecorderMux) (*handler3.OpenAPIService) {
+	w := c.RegisteredWebServices()
+
+	spec, err := builder.BuildOpenAPISpec(w, oa.Config)
+	if err != nil {
+		klog.Fatalf("Failed to build open api spec for root: %v", err)
+	}
+	// spec.Definitions = handler.PruneDefaults(spec.Definitions)
+	openAPIVersionedService, err := handler3.NewOpenAPIService(spec)
+	if err != nil {
+		klog.Fatalf("Failed to create OpenAPIService: %v", err)
+	}
+
+	err = openAPIVersionedService.RegisterOpenAPIV3VersionedService("/openapi/v3", mux)
```

Comment:
Compare https://github.com/kubernetes/enhancements/pull/2898/files#r745380166. IMO we have the chance now to get rid of all the aggregator wiring by moving the spec into the group version handler.

Reply:
I think we could even get rid of the whole apiextensions spec generation controller, and just wire it cleanly into the customresourcedefinition handler, and generate on demand.